promql/lex.go

```diff
@@ -571,7 +571,7 @@ func lexValueSequence(l *lexer) stateFn {
 		l.emit(itemBlank)
 	case unicode.IsDigit(r) || (r == '.' && unicode.IsDigit(l.peek())):
 		l.backup()
-		lexNumber(l)
+		lexNumberOrDuration(l)
```

Comment:
My read of that is that there's general support for what this change introduces, and don't think we should wait for the reworking of the go client to take advantage of it.